src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/Native/SafeHandles.cs

```diff
@@ -238,4 +238,39 @@ protected sealed override bool ReleaseHandle()
             return true;
         }
     }
+
+    internal sealed class SafeChainEngineHandle : SafeHandleZeroOrMinusOneIsInvalid
+    {
+        public SafeChainEngineHandle()
+            : base(true)
+        {
+        }
+
+        private SafeChainEngineHandle(IntPtr handle)
+            : base(true)
+        {
+            SetHandle(handle);
+        }
+
+        public static SafeChainEngineHandle MachineChainEngine =>
+            new SafeChainEngineHandle((IntPtr)ChainEngine.HCCE_LOCAL_MACHINE);
+
+        public static SafeChainEngineHandle UserChainEngine =>
+            new SafeChainEngineHandle((IntPtr)ChainEngine.HCCE_CURRENT_USER);
```

Comment:
I'm confused by these.  Did you mean to make these properties returning new instances on each call?  If so, the Dispose implementation is bad, since it a) compares this instance against new safe handle instances that it'll never equal, and b) allocates these two handles without disposing of them.  Which leads me to believe you intended these to be `=` rather than `=>`, in which case they should also be made `readonly`.